### PR TITLE
Add Boom.teapot() method with documentation and tests

### PR DESCRIPTION
**Pull request by mtharrison**
:tea:

### CODE REVIEW

**Comment by AdriVanHoudt** [comment]
:tea::shipit:

**Comment by hueniverse** [comment]
Really?

**Comment by mtharrison** [comment]
Well, it is an irreverent [joke](https://github.com/nodejs/node/blob/master/lib/_http_server.js#L58) taken maybe a bit too far. But would it really be so bad to have it? Error handling just isn't much fun and this could definitely add a smile to someone's face at just the right time. 

That being said, I would understand if you want to close it.

**Comment by hueniverse** [comment]
I don't really care, just seems like a waste of **your** time.

**Comment by devinivy** [comment]
This enhancement looks good to me.

**Comment by danivek** [comment]
+1 You made my day !

**Comment by devinivy** [comment]
@mtharrison looks like this needs a rebase.

**Comment by mtharrison** [comment]
I have a feeling this won't get merged.

**Comment by AdriVanHoudt** [comment]
have faith! :tea:

**Comment by mtharrison** [comment]
I was waiting oolong.

**Comment by jxom** [comment]
Can we please get this merged?? It is a legitimate HTTP status!!!!1111 🍵🍵🍵🍵🍵🍵

**Comment by jxom** [comment]
@hapijs pls respond

**Comment by daveols** [comment]
I'd love to see this, if not only to support RFC 2324 completely.

**Comment by lloydbenson** [comment]
This issue is closed?  Maybe open a new issue and reference this?

**Comment by AdriVanHoudt** [comment]
@arb since you are the maintainer we really need your view on this! It is tea important to let this pass imo!

**Comment by arb** [comment]
@lloydbenson side channeled my and I said "if someone opens a new PR, I'll take it."

**Comment by mtharrison** [comment]
I've rebased this, so it should be ready to go.

File: README.md [hunk 1]
```diff
@@ -511,6 +511,26 @@ Generates the following response payload:
...
 ```
 
+### `Boom.teapot([message], [data])`
+
+Returns a 418 I'm a Teapot error where:
+- `message` - optional message.
+- `data` - optional additional error data.
+
+```js
+Boom.teapot('sorry, no coffee...');
```
Comment (devinivy) [inline README.md]: Should have an upper-cased "S" on "sorry" to match the response payload below.

File: README.md [hunk 2]
```diff
@@ -32,6 +32,7 @@ Lead Maintainer: [Adam Bretz](https://github.com/arb)
     - [`Boom.unsupportedMediaType([message], [data])`](#boomunsupportedmediatypemessage-data)
     - [`Boom.rangeNotSatisfiable([message], [data])`](#boomrangenotsatisfiablemessage-data)
     - [`Boom.expectationFailed([message], [data])`](#boomexpectationfailedmessage-data)
+    - [`Boom.teapot([message], [data])`](#boomteapotmessage-data)
```
Comment (arb) [inline README.md]: Please remove. There is a tool to generate the TOC on new versions.

File: README.md (hunk 1 above)
Comment (richardlay) [inline README.md]: Awesome. It is very short...and stout

**Comment by hueniverse** [comment]
Still absurd waste of time.

**Comment by hugomd** [comment]
@hueniverse We've already wasted our time though, and it cannot be taken back!

**Comment by jxom** [comment]
@hueniverse Full support of RFC 2324 is not a waste of time! 😱
!()[https://gif.now.sh/cry/l3vRgIsVWzHkyRQkw]

**Pull request merged**